hadoop-ozone/cli-admin/src/main/java/org/apache/hadoop/hdds/scm/cli/datanode/BasicDatanodeInfoJson.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.hdds.scm.cli.datanode;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import java.util.List;
+import org.apache.hadoop.hdds.protocol.DatanodeDetails;
+import org.apache.hadoop.hdds.protocol.proto.HddsProtos;
+
+/**
+ * Represents filtered Datanode information for json use.
+ */
+
+public class BasicDatanodeInfoJson {
+  private final String id;
+  private final String hostName;
+  private final String ipAddress;
+  private final List<DatanodeDetails.Port> ports;
+  private final long setupTime;
+  private final int currentVersion;
+  private final HddsProtos.NodeOperationalState persistedOpState;
+  private final HddsProtos.NodeOperationalState opState;
+  private final long persistedOpStateExpiryEpochSec;
+  private final HddsProtos.NodeState healthState;
+  private final boolean decommissioned;
+  private final boolean maintenance;
+  private final int level;
+  private final int cost;
+  private final int numOfLeaves;
+  private final String networkFullPath;
+  private final String networkLocation;
+  private final String networkName;
+  @JsonInclude(JsonInclude.Include.NON_NULL)
+  private Long used = null;
+  @JsonInclude(JsonInclude.Include.NON_NULL)
+  private Long capacity = null;
+  @JsonInclude(JsonInclude.Include.NON_NULL)
+  private Double percentUsed = null;
+  @JsonIgnore
+  private DatanodeDetails dn;
+  
+  public BasicDatanodeInfoJson(DatanodeDetails dnDetails, HddsProtos.NodeOperationalState opState,
+      HddsProtos.NodeState healthState) {
+    this.dn = dnDetails;
+    this.id = dnDetails.getUuid().toString();
```

Comment:
We don't need to copy all the fields out of the `DatanodeDetails` here, we can just save the underlying `DatanodeDetails` object as a field. In the getters, we can forward to the getters of the DatanodeDetails. I believe Jackson should use the getters as the definitive list of fields. For example:

```
  public String getHostName() {
    return dnDetails.getHostName();
  }
```

Currently we make an unnecessary copy of every field which is never read if the `--json` flag is not passed.